course3/homework-vector/vector.cc

```diff
@@ -1,11 +1,45 @@
 #include <iostream>
 #include <cassert>
+#include <cmath>
 
 class Vector
 {
-    // TODO
+    int mX, mY;
+
+public:
+    Vector() {}
+    Vector(int x, int y) : mX(x), mY(y) {}
+
+    int GetX() const { return mX; }
+    int GetY() const { return mY; }
+
+    double GetPolarLength() const { return sqrt((mX * mX) + (mY * mY)); }
+    double GetPolarAngle() const { return atan2(mX, mY); }
+
+    void SetX(int x) { mX = x; }
+    void SetY(int y) { mY = y; }
+
+    bool operator==(const Vector& rhs) {
+        return ((rhs.GetX() == this->GetX()) && (rhs.GetY() == this->GetY()));
+    }
+
+    bool operator!=(const Vector& rhs) {
+        return !((rhs.GetX() == this->GetX()) && (rhs.GetY() == this->GetY()));
+    }
+
+    Vector operator+(const Vector& rhs) {
+        int sumX = this->GetX() + rhs.GetX();
+        int sumY = this->GetY() + rhs.GetY();
+
+        return Vector(sumX, sumY);
+    }
 };
 
+std::ostream& operator<<(std::ostream& out, const Vector& vec) {
+    return out << "{x= " << vec.GetX() << " , y= " << vec.GetY() << "}" << std::endl;
```

Comment:
Don't add ```std::endl``` inside of your streaming operator. What if your user wants to print multiple vectors in one line?